frontend/src/components/Guards/AuthGuard.js

```diff
@@ -0,0 +1,26 @@
+import React, { useContext } from "react";
+import { UserContext } from "../../context";
+import { Redirect } from "react-router-dom";
+
+function AuthGuard({children}) {
+    let {user} =  useContext(UserContext)
```

Comment:
We are using local storage to cache user data for faster retrieval.

https://github.com/codeforcauseorg/click-short/blob/e4acd91bcc071f8bcdda61c1bb780f8143661316/frontend/src/App.js#L15

The cached item don't have full information but has all the required information to know if the user is present.

use that instead of `UserContext`.
You can get that by `JSON.parse(localStorage.getItem("clickShortUser"))`.


Earlier the same logic was implemented as auth guard but using additional component for authGuard as written by you is a better idea.

Reply:
Okay I got it , I'll do this and revert back to you about this and yes using authGuard makes more sense to me that's why I tried adding this , thank you for the PR review .